db/migrate/20251124134240_add_denormalized_rdv_colmuns_to_agents_rdvs.rb

```diff
@@ -0,0 +1,26 @@
+class AddDenormalizedRdvColmunsToAgentsRdvs < ActiveRecord::Migration[8.0]
+  disable_ddl_transaction!
+
+  def change
+    add_column :agents_rdvs, :readonly_rdv_starts_at, :datetime, comment: "Colonne indexée et utilisée en lecture"
+    add_column :agents_rdvs, :readonly_rdv_ends_at, :datetime, comment: "Colonne indexée et utilisée en lecture"
+    add_column :agents_rdvs, :readonly_busy_in_the_future, :boolean, comment: "Colonne indexée et utilisée en lecture"
+
+    reversible do |direction|
+      unless Rails.env.production? # En production la migration prend trop longtemps, donc on fera tourner manuellement scripts/initialize_calculator_index_values.rb
+        direction.up do
+          AgentsRdv.update_all(<<~SQL.squish
+            readonly_rdv_starts_at = rdvs.starts_at,
+            readonly_rdv_ends_at = rdvs.ends_at,
+            readonly_busy_in_the_future = (rdvs.starts_at >= NOW() AND rdvs.status IN ('unknown', 'seen', 'noshow'))
+            FROM rdvs WHERE rdvs.id = agents_rdvs.rdv_id
+          SQL
+                              )
+        end
+      end
+    end
+
+    add_index :agents_rdvs, "agent_id, readonly_rdv_starts_at, readonly_rdv_ends_at", name: "calculator_index", algorithm: :concurrently,
```

Comment:
Ça devrait déjà nous améliorer beaucoup les perfs, mais si on veut aller plus loin je crois vraiment que le `tsrange` (avec un index R-tree) est conçu pour notre problématique. 

Si on veut vraiment digger à fond : 
- https://oxilor.com/blog/how-best-to-store-date-ranges-in-postgresql
- https://www.alibabacloud.com/blog/range-types-in-postgresql-and-gist-indexes_595128

On peut déjà merger ça puis optimiser dans le futur, c'est probablement overkill. 😉